lib/src/test/java/uk/gov/di/ipv/cri/address/library/service/AddressServiceTest.java

```diff
@@ -143,4 +151,330 @@ void shouldGetAddressItem() {
+
+        assertEquals(AddressService.ERROR_ADDRESS_LINKER_NOT_NEEDED, exception.getMessage());
+
+        assertTrue(addressService.isCurrentAddress(currentAddress));
+        assertTrue(addressService.isNotCurrentAddress(previousAddress));
+
+        // AddressProcessingException would stop this but if linker is changed...
+        // Confirm dates are not trampled over anyway
+        assertTrue((TEST_DATE_1.compareTo(previousAddress.getValidUntil()) == 0));
+        assertNotEquals(TEST_DATE_0, previousAddress.getValidUntil());
+        assertEquals(TEST_DATE_1, previousAddress.getValidUntil());
+    }
+
+    @Test
+    void testAddressesLinkerSingleCurrentAddressPasses() throws AddressProcessingException {
+
+        LocalDate date = LocalDate.of(2013, 8, 9);
+
+        CanonicalAddress currentAddress = new CanonicalAddress();
+        currentAddress.setValidFrom(date);
+
+        List<CanonicalAddress> canonicalAddresses = List.of(currentAddress);
+
+        assertTrue(addressService.isCurrentAddress(currentAddress));
+
+        addressService.setAddressValidity(canonicalAddresses);
+
+        assertTrue(addressService.isCurrentAddress(currentAddress));
+        assertEquals(date, canonicalAddresses.get(0).getValidFrom());
+    }
+
+    @Test
+    void testAddressesLinkerSingleInvalidAddressFails() {
+
+        CanonicalAddress currentAddress = new CanonicalAddress();
+        currentAddress.setValidFrom(null);
+        currentAddress.setValidUntil(null);
+
+        List<CanonicalAddress> canonicalAddresses = List.of(currentAddress);
+
+        AddressProcessingException exception =
+                assertThrows(
+                        AddressProcessingException.class,
+                        () -> addressService.setAddressValidity(canonicalAddresses));
+
+        assertFalse(addressService.isCurrentAddress(currentAddress));
+        assertEquals(AddressService.ERROR_SINGLE_ADDRESS_NOT_CURRENT, exception.getMessage());
+    }
+
+    @Test
+    void testAddressesLinkerTwoValidAddressesPass() throws AddressProcessingException {
+
+        LocalDate date = LocalDate.of(2013, 8, 9);
+
+        CanonicalAddress currentAddress = new CanonicalAddress();
+        currentAddress.setValidFrom(date);
+
+        CanonicalAddress previousAddress = new CanonicalAddress();
+        previousAddress.setValidFrom(null);
+        previousAddress.setValidUntil(null);
+
+        assertTrue(addressService.isCurrentAddress(currentAddress));
+        assertTrue(addressService.isNotCurrentAddress(previousAddress));
+        assertNull(previousAddress.getValidUntil());
+
+        List<CanonicalAddress> canonicalAddresses = List.of(currentAddress, previousAddress);
+
+        addressService.setAddressValidity(canonicalAddresses);
+
+        assertTrue(addressService.isCurrentAddress(currentAddress));
+        assertTrue(addressService.isNotCurrentAddress(previousAddress));
+        assertNotNull(previousAddress.getValidUntil());
```

Comment:
when we expect the addresses to be linked make an assertion(s) that the 
previousAddress validUntil == currentAddress.getValidFrom() etc

Please update all unit tests where appropriate

Reply:
Added where linking should have happened.